### PR TITLE
feat: add spam detection utility for bug report views

### PR DESCRIPTION
**Pull request by yashhzd**
## Summary

Adds a `spam_checker.py` utility module that implements spam detection logic for bug report submissions, addressing #2226.

### `website/spam_checker.py`
A standalone spam detection utility following the same layered pattern as `duplicate_checker.py`:

**Checks implemented:**
- **URL density analysis** - Flags submissions with excessive links (>3 URLs)
- **Spam keyword pattern matching** - Regex-based detection of common spam phrases (buy now, free money, crypto scams, etc.)
- **Account age verification** - New accounts (<7 days) and anonymous users get extra scrutiny
- **Rapid submission detection** - Flags users/IPs submitting multiple reports within 5 minutes
- **Content quality checks** - Short descriptions, ALL CAPS abuse, repetitive content

**API:**
```python
from website.spam_checker import calculate_spam_score

result = calculate_spam_score(
    description="...",
    markdown_description="...",
    user=request.user,
    reporter_ip=get_client_ip(request),
)
# result = {"score": 5, "is_spam": True, "reasons": ["High URL density", ...]}
```

### `website/tests/test_spam_checker.py`
Comprehensive test suite covering all detection functions and edge cases.

## Integration with IssueCreate view

The spam check should be added in `website/views/issue.py` inside `IssueCreate.form_valid()`, after the existing profanity check:

```python
from website.spam_checker import calculate_spam_score

# After profanity check, before rate limiting:
spam_result = calculate_spam_score(
    description=description,
    markdown_description=markdown_description,
    user=self.request.user,
    reporter_ip=reporter_ip,
)
if spam_result["is_spam"]:
    messages.error(
        self.request,
        "Your submission has been flagged for review. "
        "Please ensure your report contains a valid bug description.",
    )
    return render(self.request, "report.html", {"form": self.get_form()})
```

## Design decisions
- Scores are additive (each check adds 1-2 points) with a threshold of 3
- Flagged submissions get a user-friendly error rather than silent rejection
- Modular design allows easy addition of new spam indicators
- Pre-compiled regex patterns for performance
- Follows existing codebase conventions (logging, imports, error handling)

## Testing
```bash
python manage.py test website.tests.test_spam_checker
```

## Summary by CodeRabbit

* **New Features**
  * Added spam detection for bug reports that scores submissions, flags likely spam, and provides reasons (checks keywords, URL density, account age, rapid submissions, repetition and formatting).
  * Detection is resilient to edge cases and handles submission checks gracefully.

* **Tests**
  * Added comprehensive test coverage validating detection logic, scoring, flags, reasons, and error handling.

### CODE REVIEW

**Review by S3DFX-CYBER — CHANGES_REQUESTED** [review]
@yashhzd can u verify the output you have tested

**Review by S3DFX-CYBER — APPROVED** [review]
LGTM just solve some issues requested by cr

**Comment by yashhzd** [comment]
@S3DFX-CYBER Thanks for the review! The CI tests show all 354 tests passing (P:354 F:0 S:0) including the new spam checker tests. The CodeRabbit issues from the first review were already addressed in commit ec86823:
- Moved `Issue` import to module level (fixes the mock target issue)
- The `check_rapid_submissions` function already short-circuits when `reporter_ip` is falsy (`elif reporter_ip:` guard on line 79)

File: website/tests/test_spam_checker.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        """When reporter_ip is None, should return False without querying the DB."""
+        user = MagicMock()
+        user.is_authenticated = False
+        self.assertFalse(check_rapid_submissions(user, None))
+
+    def test_anonymous_user_with_empty_ip(self):
+        """When reporter_ip is empty string, should return False."""
+        user = MagicMock()
+        user.is_authenticated = False
+        self.assertFalse(check_rapid_submissions(user, ""))
```
[the remark this answers is not in the document]
Comment (S3DFX-CYBER) [reply website/tests/test_spam_checker.py]: Pls fix this issue

**Comment by yashhzd** [comment]
@S3DFX-CYBER This was already fixed in commit ec86823 — the `Issue` import was moved from a local import inside `check_rapid_submissions()` to a module-level import at the top of `spam_checker.py`. With that change, `@patch("website.spam_checker.Issue")` in the tests correctly intercepts the import now. The CodeRabbit comment itself also shows "✅ Addressed in commit ec86823" at the bottom. Let me know if you're seeing something else that still needs fixing!

**Comment by yashhzd** [comment]
@S3DFX-CYBER Thanks for catching that. The mock patch target `@patch("website.spam_checker.Issue")` doesn't intercept the local import inside the function. I'll fix the test to patch at the correct import path so the mock actually takes effect. Pushing a fix now.

**Comment by S3DFX-CYBER** [comment]
@coderabbitai can u summarise the changes
